tests/src/testcommon.hh

```diff
@@ -187,139 +186,28 @@ template <typename NonLinearOperator>
-      const auto expectedStressComponent = expectedStress[c][voigtIndex];
-      const bool isStressCorrect
-          = Dune::FloatCmp::eq<double, Dune::FloatCmp::CmpStyle::absolute>(FEStressComponent, expectedStressComponent);
-      t.check(isStressCorrect) << "Stress component " << voigtIndex << " at node " << c << " is not correctly computed"
-                               << messageIfFailed;
-    }
-  }
-  return t;
-}
-
-template <typename NonLinearOperator, typename FiniteElement>
-[[nodiscard]] auto checkLinearStressOfSimplex(NonLinearOperator& nonLinearOperator, FiniteElement& fe,
-                                              const std::string& messageIfFailed = "") {
-  constexpr int dim = FiniteElement::Traits::mydim;
-
-  static_assert(std::remove_cvref_t<decltype(fe.localView().tree())>::degree() == dim,
-                "The test to check linear stress is only supported with the powerBasis being two-dimensional");
-  assert(fe.localView().element().type() == Dune::GeometryTypes::simplex(dim)
-         && "The test to check linear stress is only supported for simplices");
-  static_assert(
-      std::is_same_v<
-          std::remove_cvref_t<decltype(fe.localView().tree().child(0))>,
-          Dune::Functions::LagrangeNode<std::remove_cvref_t<decltype(fe.localView().globalBasis().gridView())>, 1>>,
-      "The test to check linear stress is only supported for a linear Lagrange basis");
-
-  Dune::TestSuite t("Linear Stress check of" + std::to_string(dim) + "D  simplex element");
-
-  Eigen::Vector<double, dim*(dim + 1) / 2> expectedStress;
-  auto& displacement = nonLinearOperator.firstParameter();
-
-  if constexpr (dim == 2) {
-    displacement << 0, 0, 2, 0, 1, 0;
-    expectedStress << 2197.8021978, 659.3406593, 384.61538461;
-  } else {
-    displacement << 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0;
-    expectedStress << 576.923076900000, 1346.15384610000, 576.923076900000, 0, 384.615384600000, 769.230769200000;
-  }
-
-  using namespace Ikarus;
-  auto gridView = fe.localView().globalBasis().gridView();
-  auto element  = elements(gridView).begin();
-
-  auto resultRequirements = ResultRequirements()
-                                .insertGlobalSolution(FESolutions::displacement, displacement)
-                                .addResultRequest(ResultType::linearStress);
-
-  ResultTypeMap result;
-  fe.calculateAt(resultRequirements, Dune::FieldVector<double, dim>(0), result);
-  auto computedResult = result.getResult(ResultType::linearStress);
-
-  const bool isStressCorrect = isApproxSame(computedResult, expectedStress, 1e-8);
-  t.check(isStressCorrect) << "Computed Stress is not the same as expected stress:\n"
-                           << "It is\n:" << computedResult << "\nBut should be:\n"
-                           << expectedStress << "\n"
+  Dune::TestSuite t("Test of the calulateAt function");
+  const bool isStressCorrect = isApproxSame(computedResults, expectedResult, 1e-8);
```

Comment:
can you rename to isResultCorrect.

Furthermore, in the error message you can write somewhere `toString(resultRequirements.getRequestedResult()` to have a better error message, because then you know the actual result type that failed.